service/src/main/java/io/techery/janet/AsyncActionService.java

```diff
@@ -168,47 +193,52 @@ private void connect(boolean reconnectIfConnected) throws AsyncServiceException
         }
     }
 
-    private void disconnect() throws AsyncServiceException {
+    private void onMessageReceived(Message message) {
+        Throwable extractError = null;
+        BytesArrayBody payloadBody = null;
         try {
-            client.disconnect();
-        } catch (Throwable t) {
-            throw new AsyncServiceException(t);
-        }
-    }
-
-    private void onMessageReceived(String event, BytesArrayBody body) {
-        if (!actionsRoster.containsEvent(event)) {
-            System.err.println(String.format("Received sync message %s is not defined by any action :(. The message contains body %s", event, body));
-            return;
+            payloadBody = extractPayload(message);
```

Comment:
also, it seems like we could put extraction logic back to IncomingMessageProcessor?